assets/sharedClientFunctions.js

```diff
@@ -1,149 +0,0 @@
-        }
-      });
-    };
-    $scope.deleteRef = function (id) {
-      var url = "/reference/delete";
-      io.socket.post(url, {refId: id}, function (data, res) {
-        if (res.statusCode === 200) {
-          $scope[current_user].references = $scope[current_user].references.filter(function (ref) {
-            return ref.id !== id;
-          });
-          $scope.$apply();
-        } else {
-          console.log(res.statusCode + ": " + data);
-        }
-      });
-    };
-    $scope.isNotNormalTrade = referenceService.isNotNormalTrade;
-    $scope.hasNumber = referenceService.hasNumber;
-    $scope.isEvent = referenceService.isEvent;
-    $scope.isShiny = referenceService.isShiny;
-    $scope.isCasual = referenceService.isCasual;
-    $scope.isEggCheck = referenceService.isEggCheck;
-    $scope.isTrade = referenceService.isTrade;
-    $scope.isInvolvement = referenceService.isInvolvement;
-    $scope.isEgg = referenceService.isEgg;
-    $scope.isBank = referenceService.isBank;
-    $scope.isGiveaway = referenceService.isGiveaway;
-    $scope.isEggCheck = referenceService.isEggCheck;
-    $scope.isMisc = referenceService.isMisc;
-    $scope.isApproved = referenceService.isApproved;
-    $scope.getRedditUser = referenceService.getRedditUser;
-    $scope.formattedName = flairService.formattedName;
-    $scope.inPokemonTradesTrader = flairService.inPokemonTradesTrader;
-    $scope.inPokemonTradesHelper = flairService.inPokemonTradesHelper;
-    $scope.inSVExchangeHatcher = flairService.inSVExchangeHatcher;
-    $scope.inSVExchangeGiver = flairService.inSVExchangeGiver;
-    $scope.getFlair = flairService.getFlair;
-    $scope.userHasFlair = function (flair) {
-      return flairService.userHasFlair($scope.user, flair);
-    };
-    $scope.numberOfTrades = function () {
-      return referenceService.numberOfTrades($scope[current_user]);
-    };
-    $scope.numberOfPokemonGivenAway = function () {
-      return referenceService.numberOfPokemonGivenAway($scope[current_user]);
-    };
-    $scope.numberOfEggsGivenAway = function () {
-      return referenceService.numberOfEggsGivenAway($scope[current_user]);
-    };
-    $scope.numberOfEggChecks = function () {
-      return referenceService.numberOfEggChecks($scope[current_user]);
-    };
-    $scope.numberOfApprovedEggChecks = function () {
-      return referenceService.numberOfApprovedEggChecks($scope[current_user]);
-    };
-    $scope.getFlairTextForSVEx = function () {
-      return flairService.getFlairTextForSVEx($scope[current_user]);
-    };
-    $scope.applied = function (flair) {
-      return flairService.applied($scope.user, flair);
-    };
-    $scope.canUserApply = function (applicationFlair) {
-      return flairService.canUserApply($scope.user, applicationFlair || $scope.selectedFlair, $scope.flairs);
-    };
-    $scope.formattedRequirements = function (flair) {
-      return flairService.formattedRequirements(flair, $scope.flairs);
-    };
-  }
-};
```

Comment:
I didn't end up using a service because a lot of the shared functions require other scope variables. But I moved everything back to the shared file, so now it's not tied to a single controller anymore.